triedb/pathdb/database.go

```diff
@@ -398,14 +398,34 @@ func (db *Database) Recoverable(root common.Hash) bool {
 		return false
 	}
 	// Ensure the requested state is a canonical state and all state
-	// histories in range [id+1, disklayer.ID] are present and complete.
-	return checkHistories(db.freezer, *id+1, dl.stateID()-*id, func(m *meta) error {
-		if m.parent != root {
-			return errors.New("unexpected state history")
+	// histories in range [id+1, diskLayer.ID] are present and linked.
+	var (
+		start = *id + 1
+		limit = dl.stateID()
+	)
+	for {
+		// Short circuit if nothing more to check
+		if start > limit {
+			break
 		}
-		root = m.root
-		return nil
-	}) == nil
+		// Cap the meta items for single retrieval
+		count := limit - start + 1
+		if count > 1024 {
+			count = 1024
+		}
```

Comment:
It _can_ be done like this, but there is a more organic way of doing it. Instead of hardcoding "max 1024 items", you can simply modify `ReadStateHistoryMetaList` from
```golang
	return db.AncientRange(stateHistoryMeta, start-1, count, 0)
```
into
```golang
	return db.AncientRange(stateHistoryMeta, start-1, count, 5 * 1025 * 1025)
```
This would limit the reading to `5MB`. 
```
	// AncientRange retrieves multiple items in sequence, starting from the index 'start'.
	// It will return
	//   - at most 'count' items,
	//   - if maxBytes is specified: at least 1 item (even if exceeding the maxByteSize),
	//     but will otherwise return as many items as fit into maxByteSize.
	//   - if maxBytes is not specified, 'count' items will be returned if they are present
```
Either way is fine I guess